src/stat.h

```diff
@@ -483,10 +483,10 @@ template<class NUM> class MinValMax
   // happens when results are moved from a faster series to a slower one.
   MinValMax& operator+=(const MinValMax& rhs)
     {
-      //if (rhs.max > max) max=rhs.max;
-      //if (rhs.min < min) min=rhs.min;
-      max += rhs.max;
-      min += rhs.min;
+      if (rhs.max > max) max=rhs.max;
+      if (rhs.min < min) min=rhs.min;
+//      max += rhs.max;
+//      min += rhs.min;
       val += rhs.val;
```

Comment:
That will not work, and should not be necessary.  If you recall, 'val' is the accumulation of the data samples, such that the avg=val/samples, so it should nearly always be greater than max.  

Min and max had been the accumulation of the min and max of each sub-interval and they were then divided by the sample count.  This resulted in the published min/max values to be the average of the min/max values of the smallest intervals instead of the actual min/max values of the whole interval.